platform/fabric/core/generic/chaincode/discovery.go

```diff
@@ -127,24 +147,18 @@ func (d *Discovery) send() (discovery.Response, error) {
 		}
 	}()
 
-	//var collectionNames []string
-	//if len(d.ImplicitCollections) > 0 {
-	//	for _, collection := range d.ImplicitCollections {
-	//		collectionNames = append(collectionNames, fmt.Sprintf("_implicit_org_%s", collection))
-	//	}
-	//}
-
+	// New discovery request for:
+	// - endorsers,
+	// - config, and
+	// - local peers (for alive message)
```

Comment:
why do you use local peer query? You need to be a local admin to issue it by default, otherwise you get access denied. 

You can do a peer membership query that is channel scoped instead.

Reply:
Let me make sure I understand. Here I need to get the endorsers of a given chaincode and I also want their IP addresses. What's the minimum I need?

Reply:
The endorser query gives you the IP as well. You don't need anything else. 
If you wish the TLS root CAs you also need the config query which you use.

Reply:
perfect, I'll update the code